crates/driver/src/tests/cases/protocol_fees.rs

```diff
@@ -50,23 +50,40 @@ async fn protocol_fee_test_case(test_case: TestCase) {
         .sell_amount(test_case.execution.solver.sell)
         .buy_amount(test_case.execution.solver.buy);
     let pool = ab_adjusted_pool(quote);
-
+    let solver_fee = test_case.execution.driver.sell / 100;
+    let executed = match test_case.order.side {
+        order::Side::Buy => {
+            if test_case.order.buy_amount > test_case.execution.solver.buy {
+                Some(test_case.execution.solver.buy)
+            } else {
+                None
+            }
```

Comment:
Syntactic 🍬 before @MartinquaXD beats me to it.

```suggestion
            (test_case.order.buy_amount > test_case.execution.solver.buy).then_some(test_case.execution.solver.buy)
```

Reply:
Lol, initially used this syntax, but found it hard to read and reverted to explicit if/else.